logger.go

```diff
@@ -0,0 +1,87 @@
+package tracelog
+
+import (
+	"fmt"
+	"os"
+)
+
+type LoggerType string
+
+const (
+	InfoLoggerType    LoggerType = "INFO"
+	WarningLoggerType LoggerType = "WARNING"
+	ErrorLoggerType   LoggerType = "ERROR"
+	DebugLoggerType   LoggerType = "DEBUG"
+)
+
+type Fields map[string]interface{}
+type FieldValues func() Fields
+
+type Logger struct {
+	loggerWriters []LoggerWriter
+	fieldValues   FieldValues
+}
+
+func NewLogger(fieldValues FieldValues, loggerWriters ...LoggerWriter) *Logger {
```

Comment:
We don't need many writers at the same time, so I suggest to change it to a single writer for simplicity.